src/realm/cluster.cpp

```diff
@@ -74,14 +74,14 @@ inline void Cluster::do_create(ColKey col)
     arr.update_parent();
 }
 
-void Cluster::create(size_t nb_leaf_columns)
+void Cluster::create()
 {
-    // Create array with the required size
-    Array::create(type_HasRefs, false, nb_leaf_columns + s_first_col_index);
+    Array::create(type_HasRefs, false, s_first_col_index);
     Array::set(0, RefOrTagged::make_tagged(0));
-    auto table = m_tree_top.get_owner();
     auto column_initialize = [this](ColKey col_key) {
         auto col_ndx = col_key.get_index();
+        while (size() <= col_ndx.val + 1)
```

Comment:
don't we have a way of resizing to the proper size at once. This may involve multiple underlying resize, free and alloc operations as well as copy overhead.

Reply:
When we COW, we allocate an extra 64 bytes, and on subsequent re-alloc we double the size each time, so I would not expect this to be a huge problem. We could of course add another virtual function that would give us the needed size.